src/odemis/gui/comp/overlay/world.py

```diff
@@ -832,6 +833,566 @@ def deactivate(self):
         WorldOverlay.deactivate(self)
 
 
+NONE_RULER_MODE = 0
+MOVE_RULER_MODE = 1
+EDIT_START_RULER_MODE = 2
+EDIT_END_RULER_MODE = 3
+
+class Ruler(object):
+
+    hover_margin = 10  # pixels
+
+    def __init__(self, cnvs, drag_v_start_pos, drag_v_end_pos):
+
+        self.cnvs = cnvs
+
+        self.colour = conversion.hex_to_frgba(gui.CROSSHAIR_COLOR) # green colour
+        self.highlight = conversion.hex_to_frgba(gui.FG_COLOUR_HIGHLIGHT)  # orange colour for the selected ruler
+
+        self.ruler_label = Label(
+            "",
+            pos=(0, 0),
+            font_size=12,
+            flip=True,
+            align=wx.ALIGN_CENTRE_HORIZONTAL,
+            colour=self.colour,
+            opacity=1.0,
+            deg=None
+        )
+
+        self.drag_v_start_pos = drag_v_start_pos
```

Comment:
Add a comment like: "positions (in view coordinates) where the drag started, and where it is now".